Use new onboarding flow throughout the app

mchowning opened the pull request:
| 📘 Project: #524  |
| --- |

## Description
This PR is bringing the screens from the onboarding flow to the rest of the app.

Sorry for the big PR. These flows were so interrelated that changing one could break any of the other ones so I thought it would be easier to just test them all at once as opposed to repeatedly re-testing them across multiple PRs.

## Testing Instructions

Test that the appropriate flows start from the appropriate location. With each flow, also verify that an appropriate "source" property is sent to Tracks when the Plus Upgrade screen is presented during the flow.

In parentheses, I note the flow tag that should be sent to tracks along with the events from that flow, i.e.: `setup_account_shown, Properties: {"flow":"plus_upsell"...`.

### A. Full onboarding flow

####  Locations

1.  First time the user opens the app after a new installation ("initial_onboarding" flow)
2.  Profile → Sign In ("logged_out" flow)

####  Flow

1.  Login or Sign Up
2.  If Login, exit flow, otherwise
4.  If created a new account: Recommendations (going back from here should exit the flow)
5.  Plus Upgrade
6.  Welcome (if the user upgraded, going back from here should exit the flow)

### B. Plus Upgrade flow

This is what we send if the user is indicating they want to upgrade (i.e., clicking an Upgrade button). Because the user is indicating they want upgrade already, we skip the Plus Upgrade\'s full screen view and immediately open the purchase bottom modal.

####  Locations

1. Profile tab → \"Signed in as\" box → Tap on upgrade/start-trial button
2. Profile tab → ⚙ → Plus Details → Upgrade Button

####  Signed In ("plus_account_upgrade" flow)

1.  Purchase modal
2.  After exiting the flow, the upgrade screen should be closed and the user should be returned to the settings screen

####  Signed Out ("plus_account_upgrade_needs_login" flow)

1.  Login or Sign up
2.  If created a new account: Recommendations
3.  Purchase bottom modal (dismissing the modal presents the full screen upsell page, then tapping Not Now or swiping back will exit the flow)
4.  If created a new account: Welcome. If logged into a previous account: exit the flow.
5.  After exiting the flow, the upgrade screen should be closed and the user should be returned to the settings screen

### C. Plus Upsell flow ("plus_upsell" flow)

This is what we send if the user has tapped on a Plus feature or
otherwise indicated they are interested in learning more about
upgrading.

####  Locations

1.  Profile tab → ⚙ → Appearance → Plus Theme
2.  Profile tab → ⚙ → Appearance → Plus Icon
3.  Profile tab → ⚙ → Appearance → Plus banner at bottom of screen
4.  Profile tab → Files → Add File → Image Placeholder
5.  Profile tab → Files → Add File → \"Add Custom Image\"
7.  Profile tab → Files → Add File → Tapping on a color with a 🔒
8.  Profile tab → Files → Add File → Plus banner at bottom of screen
9.  Profile tab → Files → Overflow menu → File settings → Tap on one of the Plus features
10.  Profile tab → Files → Overflow menu → File settings → Plus banner at bottom of screen
11. Profile tab → \"Signed in as\" box → Learn more about Pocket Casts
12. Profile tab → Plus banner at bottom of screen
13. Profile tab → \"Signed in as\" box → Start Free Trial / Upgrade button
14. Podcasts tab → Folders upgrade prompt

#### Signed In

1.  Plus Upgrade Full Screen
2.  Tapping button to upgrade presents purchase modal (alternatively, tapping Not Now or swiping back exits the flow)
3.  Welcome Screen

####  Signed Out

1.  Plus Upgrade Full Screen
2.  Tapping button to upgrade presents the Login or Sign Up flow
3.  If created a new account: Recommendations
4.  Plus Upgrade screen with the purchase modal automatically presented (tapping back will dismiss the purchase modal and present the full-screen plus upsell page)
5.  If created a new account: Welcome Screen (if you are in one of the File upsell flows, the Go to DIscover option will not be presented on this page. That was done for two reasons: (1) the user was in the middle of adding a file, so this avoids aborting that process and (2) navigating the user back between activities to the discover screen programmatically would be tricky and fragile to implement).

## Screenshots or Screencast 
This is what the full onboarding flow looks like. Note that the particular flow for a particular part of the app will likely be slightly different from this as described above in the Testing section.

https://user-images.githubusercontent.com/4656348/211891196-480d8de7-d2c9-43ba-8e42-b73884a2fdf5.mov


## Checklist
- [X] If this is a user-facing change, I have added an entry in CHANGELOG.md
- [X] I have considered whether it makes sense to add tests for my changes
- [X] All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`
- [X] Any jetpack compose components I added or changed are covered by compose previews
 
#### I have tested any UI changes...
<!-- If this PR does not contain UI changes, ignore these items -->
- [X] with different themes
- [X] with a landscape orientation
- [X] with the device set to have a large display and font size
- [X] for accessibility with TalkBack

mchowning commented on modules/features/settings/src/main/java/au/com/shiftyjelly/pocketcasts/settings/AppearanceThemeSettingsAdapter.kt:
```diff
@@ -108,11 +108,8 @@ class AppearanceThemeSettingsAdapter(
-            var afterTheme = list[bindingAdapterPosition]
+            val afterTheme = list[bindingAdapterPosition]
             val validTheme = !afterTheme.isPlus || isPlusSignedIn
-            if (afterTheme.isPlus && !isPlusSignedIn) {
-                afterTheme = beforeTheme // prevent changing theme
-            }
```
I removed this check (and the identical check in `AppearanceThemeSettingsAdapter` because:

1. These checks were added in order to make sure that a users selection appropriately updated to an allowed theme when they tried to select a plus theme but did not have a plus account (see 2062-gh-automattic/pocket-casts-android-private), but I do not see that issue occurring even with the checks removed.

3. There is code in place making it so that a user who (1) taps on a plus theme/icon, (2) upgrades to plus in the upgrade prompt that pops up, will then (3) have the plus icon/theme they tapped on activated automatically, but this was broken by these checks. So removing those checks fixed that behavior.

ashiagr commented:
Kudos for writing detailed testing steps, @mchowning!

Here are my notes from testing:

With each flow, also verify that an appropriate "source" property is sent to Tracks when the Plus Upgrade screen is presented during the flow.  ✅ 

 ### A. Full onboarding flow ✅ 

### B. Plus Upgrade flow 
####  Locations
2. Profile tab → ⚙ → Plus Details → Upgrade Button ❓ 

I was not able to dismiss the Plus Upgrade flow here if I was logged in with a non-license testing account. I couldn't reproduce it with a license-testing account, just noting it down in case it can be fixed easily:

https://user-images.githubusercontent.com/1405144/212006713-669c7270-0edb-46ba-8282-6176ca29495d.mp4

### C. Plus Upsell flow ("plus_upsell" flow)

####  Locations
8.  Profile tab → Files → Add File → Plus banner at bottom of screen ❌ 
I can only see `Plus Upsell flow` from `Learn More` link but not from the `Plus banner` here.

https://user-images.githubusercontent.com/1405144/212004066-470b4f53-b6c4-43a7-acb7-905f933ac651.mp4

####  Signed Out

5.  If created a new account: Welcome Screen (if you are in one of the File upsell flows, the Go to DIscover option will not be presented on this page. That was done for two reasons: (1) the user was in the middle of adding a file, so this avoids aborting that process and (2) navigating the user back between activities to the discover screen programmatically would be tricky and fragile to implement). - Makes sense 👍

I also noticed two issues that are not related to changes in this PR but are worth fixing:

1. If there's no free trial offer, the Upgrade screen still displays the `Start Free Trial & Subscribe` button.

<img src="https://user-images.githubusercontent.com/1405144/212004590-096e094c-2f17-4071-99d4-eaef2b2f8f0b.png" width=320/>

2. On the Pocket Casts Plus page, I see yearly trial text for a monthly trial:

<img src="https://user-images.githubusercontent.com/1405144/212005035-129f37f3-1d3e-4ab9-bf90-3553b2b80984.png" width=320/>

Hope this helps. Great job in covering all scenarios! 🌟

mchowning commented:
Thanks so much for the careful review @ashiagr ! I know this wasn't a fun one to review, and I really appreciate you going through all the flows and catching some issues. 🙇 

> Profile tab → ⚙ → Plus Details → Upgrade Button ❌ 
> I was not able to dismiss the Plus Upgrade flow here

Fixed in https://github.com/Automattic/pocket-casts-android/pull/694/commits/60ab9849cdba15831d2d4d802798be048a1c5cf3

> Profile tab → Files → Add File → Plus banner at bottom of screen ❌
> I can only see Plus Upsell flow from Learn More link but not from the Plus banner here.

Made the entire banner clickable in https://github.com/Automattic/pocket-casts-android/pull/694/commits/d4fbd175016018f99c0c08599f028e8b0f659906

> If there's no free trial offer, the Upgrade screen still displays the Start Free Trial & Subscribe button.

Oh that's not good at all. Feels like this should be a beta-fix, so I'll address it in a separate PR.

> On the Pocket Casts Plus page, I see yearly trial text for a monthly trial

Good catch. Fixed in https://github.com/Automattic/pocket-casts-android/pull/694/commits/837c2dc8970692f49294a577e8c6fa98c43b94b9

ashiagr reviewed (approved):
Thank you for the updates! All looks good to me.

mchowning commented:
Retested after merging in `main` to verify that SSO still works, and I confirmed that I can still log in with SSO.
